WooCommerce/Classes/System/WidgetExtensionService.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+import KeychainAccess
+import Networking
+
+// WidgetExtensionService class
+// Save, Load and Remove credentials used by Extensions of the Woo app, like the TodayStatsWidget
+//
+final class WidgetExtensionService: NSObject {
+
+    /// Returns the Credentials, if any, under the related App Groups.
+    ///
+    static func loadCredentials() -> Credentials? {
```

Comment:
I'm also concerned that we have `WidgetExtensionService` injected everywhere. I believe this makes it less clear how the Widgets target depends on the WooCommerce, Yosemite, and other frameworks. 

I haven't fully grasped this PR yet but I wonder if it's possible to make it look like this:

![image](https://user-images.githubusercontent.com/198826/85641511-02dc8e80-b64c-11ea-84d7-595215de673b.png)

### WidgetCredentialsRepository

- This is shared by the WooCommerce and the Widgets Targets. 
- Encapsulates the `UserDefaults` magic.  
- Has no static methods. It has to be instantiated. It can be kept as a private instance of its consumer.
- Maybe this can handle validation of sorts that both the `Credentials` and `Site` must exist together or the "complete credentials" is invalid/unusable. 

### SessionManager

- Dispatches change events whenever the `site` and the `credentials` are updated using `PublishSubject` + `Observable`. 
- No other code should be added here so that its only concern is keeping information about the current session. 

### WidgetCredentialsUpdater

- Belongs in the WooCommerce Target only.
- Upon activation, observes the Observables provided by `SessionManager` (Credentials and Site)
- Uses an instance of `WidgetCredentialsRepository`
- Updates `WidgetCredentialsRepository` whenever `SessionManager`'s Observables emit new changes.
- This can be kept by `AppDelegate` probably so it's not accessible by other classes. But it has to be active in memory in order to work.

### TodayViewController

- Uses an instance of `WidgetCredentialsRepository` to read the credentials and site. 

---

If we can implement it this way, we reverse the dependency. Instead of `SessionManager` **depending** on `WidgetExtensionService`, it's now the other way around. This also makes it easier to take out and refactor the Widget system. For example, to remove the widgets from WooCommerce target:

1. Delete the `WidgetCredentialsUpdater`
2. That's it. Well, except the `UserDefaults` keys I guess.